hazelcast/src/main/java/com/hazelcast/util/PhoneHome.java

```diff
@@ -204,6 +218,164 @@ private void fetchWebService(String urlStr) {
         }
     }
 
+    private void addOSInfo(PhoneHomeParameterCreator parameterCreator) {
+        OperatingSystemMXBean osMxBean = ManagementFactory.getOperatingSystemMXBean();
+        try {
+            parameterCreator.addParam("osn", osMxBean.getName());
+            parameterCreator.addParam("osa", osMxBean.getArch());
+            parameterCreator.addParam("osv", osMxBean.getVersion());
+        } catch (SecurityException e) {
+            parameterCreator.addParam("osn", "N/A");
+            parameterCreator.addParam("osa", "N/A");
+            parameterCreator.addParam("osv", "N/A");
+        }
+    }
+
+    private void addClientInfo(Node hazelcastNode, PhoneHomeParameterCreator parameterCreator) {
+        Map<ClientType, Integer> clusterClientStats = hazelcastNode.clientEngine.getConnectedClientStats();
+
+        parameterCreator.addParam("ccpp", Integer.toString(clusterClientStats.get(ClientType.CPP)));
+        parameterCreator.addParam("cdn", Integer.toString(clusterClientStats.get(ClientType.CSHARP)));
+        parameterCreator.addParam("cjv", Integer.toString(clusterClientStats.get(ClientType.JAVA)));
+        parameterCreator.addParam("cnjs", Integer.toString(clusterClientStats.get(ClientType.NODEJS)));
+        parameterCreator.addParam("cpy", Integer.toString(clusterClientStats.get(ClientType.PYTHON)));
+    }
+
+    private void addManCenterInfo(ExecutorService executor, Future<MCResponse> response,
+                                  int clusterSize, PhoneHomeParameterCreator parameterCreator) {
+        MCResponse mcResponse = null;
+        try {
+            mcResponse = response.get();
+        } catch (InterruptedException ignored) {
+            EmptyStatement.ignore(ignored);
+        } catch (ExecutionException ignored) {
+            EmptyStatement.ignore(ignored);
+        } finally {
```

Comment:
I would combine both those exceptions into `Exception` and set `MC_NOT_AVAILABLE` just here in case Exception has been thrown or `mcResponse==true`
No need to go further from here if we have problems fetching info from MC